commons/src/test/java/io/hyscale/commons/utils/HyscaleStringUtilsTests.java

```diff
@@ -15,39 +15,74 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 public class HyscaleStringUtilsTests {
-    private String sampleString = "hyscaleUser";
-    private String suffixString = "User";
-    private String expectedString= "hyscale";
+    private static final String SAMPLE_STRING = "hyscaleUser";
+    private static final String SUFFIX_STRING = "User";
+    private static final String EXPECTED_STRING = "hyscale";
+    private static final char TAILING_CHAR = 'r';
+    private static final String EXPECTED_CHAR_STRING="hyscaleUse";
+
+    public static Stream<Arguments> getStrInputs() {
+        return Stream.of(Arguments.of(null, SUFFIX_STRING, null),
+                Arguments.of("", SUFFIX_STRING, ""),
+                Arguments.of(SAMPLE_STRING, null, SAMPLE_STRING),
+                Arguments.of("", SUFFIX_STRING, ""),
+                Arguments.of(SAMPLE_STRING, SUFFIX_STRING, EXPECTED_STRING));
+    }
 
-    @Test
-    public void testRemoveSuffixStr() {
+    @ParameterizedTest
+    @MethodSource(value = "getStrInputs")
+    public void testRemoveSuffixStr(String sampleString, String suffixString, String expected) {
         String actualString = HyscaleStringUtil.removeSuffixStr(sampleString, suffixString);
-        assertNotNull(actualString);
-        assertEquals(expectedString, actualString);
+            assertEquals(expected, actualString);
     }
 
-    @Test
-    public void testRemoveSuffixStrBuilder() {
-        StringBuilder str = new StringBuilder();
-        str.append(sampleString);
-        String actualString = HyscaleStringUtil.removeSuffixStr(str, suffixString);
-        assertNotNull(actualString);
-        assertEquals(expectedString, actualString);
+    public static Stream<Arguments> getStringBuilderInputs() {
+        return Stream.of(Arguments.of(null, SUFFIX_STRING, null),
+                Arguments.of(SAMPLE_STRING, null, SAMPLE_STRING),
+                Arguments.of(SAMPLE_STRING, " ", SAMPLE_STRING),
+                Arguments.of(SAMPLE_STRING, SUFFIX_STRING, EXPECTED_STRING));
     }
 
-    @Test
-    public void testRemoveSuffixChar() {
-        StringBuilder str = new StringBuilder();
-        str.append(sampleString);
-        String actualString = HyscaleStringUtil.removeSuffixChar(str, 'r');
-        assertNotNull(actualString);
-        assertEquals("hyscaleUse", actualString);
+    @ParameterizedTest
+    @MethodSource(value = "getStringBuilderInputs")
+    public void testRemoveSuffixStrBuilder(String sampleString, String suffix, String expected) {
+        StringBuilder str = getStringBuilderFor(sampleString);
+        String actualString = HyscaleStringUtil.removeSuffixStr(str, suffix);
+        assertEquals(expected, actualString);
+    }
```

Comment:
The same set of arguments to removeSuffixStr() with String & StringBuilder is enough?

Reply:
yes,becauser input  need to check 
null,
any string builder object input.